Content proposal for includeClass call

rakshitkr opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes Issue #343 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Eclipse Version: 2020-03
* Java Version: 1.8
* OS: Mac Catalina

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

rakshitkr committed:
Extended content proposal of Java editor to include CrySL rulenames

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

- includes proposals for CrySL class names in includeClass call

rakshitkr committed:
Fixed visit to parse current method in method chaining

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Added extension for java content proposal

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Refactored code

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Updated ASTParser

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Added methods to read only classnames from crysl rules

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Merge remote-tracking branch 'upstream/develop' into content-assist

rakshitkr committed:
Changed the ruleset location to eclipse directory

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Changed JSL13 to JSL12

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

enriozuni commented:
I tested this PR in my machine and everything works as expected.

kruegers commented on plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/ui/contentassist/SelectionFinder.java:
```diff
@@ -0,0 +1,77 @@
+	
+	
+	// Solution 2
+//	@Override
```
Please remove if no longer necessary.

kruegers commented on plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/ui/contentassist/CustomCompletionProposalComputer.java:
```diff
@@ -0,0 +1,97 @@
+
+	private static String readClassnameFromRule(File ruleFile) throws IOException {
+		final String fileName = ruleFile.getName();
+		if (!fileName.endsWith(".crysl")) {
```
Use constant in constant class for crysl file ending.

kruegers commented on plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/ui/contentassist/CustomCompletionProposalComputer.java:
```diff
@@ -0,0 +1,97 @@
+@SuppressWarnings("restriction")
+public class CustomCompletionProposalComputer extends JavaCompletionProposalComputer {
+	
+	static private String rulesDir = Constants.ECLIPSE_RULES_DIR + Constants.innerFileSeparator + Constants.Rules.JavaCryptographicArchitecture.toString() + Constants.innerFileSeparator + 
```
Would it make sense to store this as a constant in the constant class? If not, why?

rakshitkr replied:
You are right! Since both content assist and code generator both uses latest jca ruleset from eclipse path, it makes sense to move it to Constants class

kruegers reviewed (changes requested):
Thanks, Rakshit. Plase take care of the two minor things and the PR can be merged.

rakshitkr committed:
Removed redundant code

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

- deleted solution 2
- decided to go with solution 1 instead

rakshitkr committed:
Moved JCA latest ruleset eclipse path to Constants

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

- created a new constant named JCA_LATEST_ECLIPSE_RULES_DIR for retrieve latest JCA ruleset
- used it for content proposal & code generation

rakshitkr commented:
@kruegers I've made all the requested changes and tested the modified code. You can go ahead with the merge
